ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py

```diff
@@ -162,3 +162,48 @@ def test_numpy_std(
         where=where,
         test_values=False,
     )
+
+
+#average
+@handle_frontend_test(
+    fn_tree="numpy.average",
+    dtype_and_a=statistical_dtype_values(function="average"),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.numpy.average"
+    ),
+    dtype_and_x=statistical_dtype_values(function="average"),
+    keep_dims=st.booleans(),
+)
+def test_numpy_average(
+        dtype_and_a,
+        as_variable,
+        with_out,
+        native_array,
+        num_positional_args,
+        dtype_and_x,
+        frontend,
+        fn_tree,
+        keep_dims,
```

Comment:
You should also add `on_device` to the arguments and pass it to `test_frontend_function` below, as done for example [here](https://github.com/unifyai/ivy/blob/2ae5f1e6b6ffd10e95c8e8c68f6747dd24cd4f0a/ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_decompositions.py#L27).